### PR TITLE
[KOGITO-3463] - Update clone-repo script for CI specific actions

### PR DESCRIPTION
**Pull request by Kaitou786**
The motive of this PR is to not overwrite the settings.xml file when running from a local machine
See: https://issues.redhat.com/browse/KOGITO-3463

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -21,6 +21,9 @@ pipeline{
     tools {
         jdk 'kie-jdk11'
     }
+    environment {
+        CI = true
```
Comment (radtriste) [inline Jenkinsfile:25]: that works ? AFAIK, environment var in Jenkinsfile should be strings ...

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -59,6 +59,7 @@ pipeline {
     }
 
     environment {
+        CI = true
```
Comment (radtriste) [inline Jenkinsfile.deploy:62]: same here

File: Jenkinsfile.promote [hunk 3]
```diff
@@ -61,6 +61,7 @@ pipeline {
     }
 
     environment {
+        CI = true
```
Comment (radtriste) [inline Jenkinsfile.promote:64]: same here

**Review by radtriste — CHANGES_REQUESTED** [review]
Could you make the title more specific ? Like `Updating clone-repo script for CI specific actions`

File: Jenkinsfile (hunk 1 above)
Comment (Kaitou786) [reply Jenkinsfile:25]: Yes, checked in this [URL](https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/kaitou786/job/forkRepo%20test/17/replay/)

Comment (radtriste) [reply Jenkinsfile:25]: oki :)

File: tests/test-apps/clone-repo.sh [hunk 4]
```diff
@@ -9,13 +9,15 @@ MAVEN_QUARKUS_NATIVE_CONTAINER_BUILD_ARGS="-Dquarkus.native.container-build=true
...
-
+#Setup maven configuration only on CI
+if [ $CI ]; then
 # setup maven env
-export JBOSS_MAVEN_REPO_URL="https://repository.jboss.org/nexus/content/groups/public/"
-# export MAVEN_REPO_URL=
-cp ${MVN_MODULE}/maven/settings.xml ${HOME}/.m2/settings.xml
-source ${MVN_MODULE}/added/configure-maven.sh
-configure
+    export JBOSS_MAVEN_REPO_URL="https://repository.jboss.org/nexus/content/groups/public/"
```
Comment (radtriste) [inline tests/test-apps/clone-repo.sh:15]: can you check replace regex are still working in python scripts if we add the spaces at the beginning of the line ?

Comment (Kaitou786) [reply tests/test-apps/clone-repo.sh:15]: Checking..

**Review by sutaakar — APPROVED** [review]

File: tests/test-apps/clone-repo.sh (hunk 4 above)
Comment (Kaitou786) [reply tests/test-apps/clone-repo.sh:15]: Works well with the indentation as well =D

**Review by radtriste — APPROVED** [review]

**Review by spolti — APPROVED** [review]
Many thanks @Kaitou786

**Pull request merged**